apps/src/StudioApp.js

```diff
@@ -2177,7 +2177,7 @@ StudioApp.prototype.configureDom = function (config) {
     if (!runButtonWasClicked) {
       analyticsReporter.sendEvent(
         eventName,
-        {signedIn: config.isSignedIn},
+        {signedIn: config.isSignedIn, unitName: config.scriptName},
```

Comment:
Are you interested in including `scriptName` for `javalab` as well? `javalab` uses a different 'Run' button, so this event is not logged for `javalab` levels/projects even though `javalab` does use `StudioApp`.
There is a different event logged whenever the `javalab` 'Run' button is logged which includes the level id (not just the first time 'Run' button click). 
https://github.com/code-dot-org/code-dot-org/blob/4ddae9c0a70b2313be15b42dd689be9283c77fdb/apps/src/javalab/Javalab.js#L379-L380

Reply:
ooo good catch! I think this is fine for now ... for context, we're seeing some unexpected user journeys. In the past, these have been due to a bug, but I think this time it's actually real data. But my hypothesis relies on these level activity events being fired in CSF, hence wanting the unit name :)

Reply:
Oh that sounds like interesting detective work 🕵️‍♀️ ! And just a quick note that we discussed if first click of 'Run' button should be tracked for `lab2` labs as well.